src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -349,27 +350,41 @@ public synchronized void applyDML(Sequenced<DMLEvent> sequencedEvent) throws Exc
     }
 
     DMLEvent event = sequencedEvent.getEvent();
-    gcsWriter.write(sequencedEvent);
+    String normalizedDatabaseName = normalize(event.getDatabase());
+    String normalizedTableName = normalize(event.getTable());
+    DMLEvent normalizedDMLEvent = DMLEvent.builder()
+      .setOffset(event.getOffset())
```

Comment:
it will be easy to miss something if we add a new property to DMLEvent. Let's add a builder() method that takes an existing DMLEvent and sets up all the Builder fields already. This can then be changed to:

```
DMLEvent normalizedEvent = DMLEvent.builder(event)
  .setDatabase(normalizedDatabaseName)
  .setTable(normalizedTableName)
  .build()
```

That way, if we add any fields in the future, we just need to update that builder() method, we don't need to change any plugins that happen to do something like this.

Reply:
That's a good suggestion. Will do.

Reply:
PR out: https://github.com/data-integrations/delta/pull/48